dds/DCPS/DomainParticipantImpl.cpp

```diff
@@ -121,6 +121,20 @@ DomainParticipantImpl::DomainParticipantImpl(
 
 DomainParticipantImpl::~DomainParticipantImpl()
 {
+#ifdef OPENDDS_SECURITY
+  if (security_config_ && perm_handle_ != DDS::HANDLE_NIL) {
+    Security::AccessControl_var access = security_config_->get_access_control();
+    DDS::Security::SecurityException se;
+    const bool okay = access->return_permissions_handle(perm_handle_, se);
+    if (!okay) {
+      ACE_ERROR((LM_ERROR,
```

Comment:
should this log statement be conditional like the one below?

Reply:
Yes.